src/Optimizer/AmpWPConfiguration.php

```diff
@@ -0,0 +1,119 @@
+<?php
+/**
+ * Class AmpWPConfiguration.
+ *
+ * @package AmpProject\AmpWP
+ */
+
+namespace AmpProject\AmpWP\Optimizer;
+
+use AmpProject\Optimizer\Configuration;
+use AmpProject\Optimizer\DefaultConfiguration;
+use AmpProject\Optimizer\Exception\UnknownConfigurationKey;
+use AmpProject\Optimizer\Transformer;
+use AmpProject\AmpWP\Optimizer\Transformer as WpTransformer;
+use AmpProject\Optimizer\TransformerConfiguration;
+
+/**
+ * Optimizer Configuration implementation that is mutable and filterable.
+ *
+ * @package AmpProject\AmpWP
+ * @since 2.1.0
+ * @internal
+ */
+final class AmpWPConfiguration extends DefaultConfiguration {
+
+	/**
+	 * Whether the filters have already been applied.
+	 *
+	 * @var bool
+	 */
+	private $already_applied = false;
+
+	/**
+	 * Apply the filters to adapt the configuration.
+	 *
+	 * Note: They will only be applied once, when this method is hit for the first time.
+	 */
+	public function apply_filters() {
+		if ( $this->already_applied ) {
+			return;
+		}
```

Comment:
Any chance this would be undesirable?  What if you wanted to change the configuration with a filter change on a subsequent call?

Reply:
The alternative is to run the filters on every `get()` and `has()`, which can quickly become a performance issue.

If you want to change the configuration with a later filter call, you can instantiate a new configuration object instead.

Also, this should actually only be triggered when the optimizer is being run, which happens in the shutdown hook that collects the final output buffer after WordPress has finished. I don't think plugins should do anything _after_ the optimizer has run.